### PR TITLE
port commodities added as constraints for placement

### PR DESCRIPTION
**Pull request by pamelasanchezvi**
Created port commodities based on the ports used by each container at the VMs.

### CODE REVIEW

**Comment by pamelasanchezvi** [comment]
let me know if there is anything else to update, I have tested with 91 so far !

File: communicator/probe/node_probe.go [hunk 1]
```diff
@@ -90,70 +115,70 @@ func (nodeProbe *NodeProbe) CreateCommoditySold(slaveInfo *util.Slave, useMap ma
...
+			Key(port). // port number in string form
+			Capacity(portObj.Capacity).
+			Used(portObj.Used).
+			Create()
+		commoditiesSold = append(commoditiesSold, portComm)
 	}
-	return commoditiesSold, nil
+
+	// add labels
+	/*	for _, label := range labels {
```
Comment (DongyiYang) [inline communicator/probe/node_probe.go]: Why you comment out VMPM Access Comm

Comment (DongyiYang) [inline communicator/probe/node_probe.go:177]: why comment our VMPM access Commodity

File: communicator/metadata/vmt_meta.go [hunk 2]
```diff
@@ -34,9 +34,12 @@ const (
 )
 
 type VMTMeta struct {
```
Comment (DongyiYang) [inline communicator/metadata/vmt_meta.go]: I think VMTMeta is for storing the info that is used to connect to Turbo server. It shouldn't have included any field relates to Mesos, Marathon or Action

File: communicator/util/types.go [hunk 3]
```diff
@@ -15,12 +13,18 @@ type Resources struct {
 	Ports string  `json:"ports"`
 }
 
+type PortUtil struct {
```
Comment (DongyiYang) [inline communicator/util/types.go:16]: Add some comment about what is this **PortUtil**.

File: communicator/vmturbocommunicator/server_message_handler.go [hunk 4]
```diff
@@ -264,15 +266,15 @@ func (handler *MesosServerMessageHandler) HandleAction(serverMsg *comm.Mediation
 }
 
 func (handler *MesosServerMessageHandler) NewMesosProbe(previousUseMap map[string]*util.CalculatedUse) (*util.MesosAPIResponse, error) {
```
Comment (DongyiYang) [inline communicator/vmturbocommunicator/server_message_handler.go:268]: This is a giant method. Please extract code to new methods or functions.

File: communicator/vmturbocommunicator/vmt_communicator.go [hunk 5]
```diff
@@ -114,20 +113,19 @@ func createSupplyChain() []*sdk.TemplateDTO {
 		Entity(sdk.EntityDTO_VIRTUAL_MACHINE).
 		Selling(sdk.CommodityDTO_CPU_ALLOCATION, fakeKey).
 		Selling(sdk.CommodityDTO_MEM_ALLOCATION, fakeKey).
-		Selling(sdk.CommodityDTO_STORAGE_ALLOCATION, fakeKey).
+		//		Selling(sdk.CommodityDTO_STORAGE_ALLOCATION, fakeKey).
```
Comment (DongyiYang) [inline communicator/vmturbocommunicator/vmt_communicator.go:116]: why comment out?

File: communicator/vmturbocommunicator/vmt_communicator.go [hunk 6]
```diff
@@ -141,18 +139,22 @@ func createSupplyChain() []*sdk.TemplateDTO {
...
-		Key:           &fakeKey,
-		CommodityType: &diskAllocationType,
-	}
-	clusterType := sdk.CommodityDTO_CLUSTER
+	/*	diskAllocationType := sdk.CommodityDTO_STORAGE_ALLOCATION
+		diskAllocationTemplateComm := &sdk.TemplateCommodity{
+			Key:           &fakeKey,
+			CommodityType: &diskAllocationType,
+		}
+	*/clusterType := sdk.CommodityDTO_CLUSTER
```
Comment (DongyiYang) [inline communicator/vmturbocommunicator/vmt_communicator.go]: move clusterType to a new line!

File: communicator/vmturbocommunicator/vmt_communicator.go [hunk 7]
```diff
@@ -162,7 +164,7 @@ func createSupplyChain() []*sdk.TemplateDTO {
 		Provider(sdk.EntityDTO_VIRTUAL_MACHINE, sdk.Provider_LAYERED_OVER).
 		Buys(*cpuAllocationTemplateComm).
 		Buys(*memAllocationTemplateComm).
-		Buys(*diskAllocationTemplateComm).
+		//		Buys(*diskAllocationTemplateComm).
```
Comment (DongyiYang) [inline communicator/vmturbocommunicator/vmt_communicator.go:169]: We don't need diskAllocationCommodity?

File: mesosturbo/main.go [hunk 8]
```diff
@@ -12,51 +12,72 @@ import (
...
 var opsman_password string
+var action_api string
+var action_ip string
+var action_port string
+var slave_port string
+var username string
+var password string
+var targetid string
 
 func init() {
```
Comment (DongyiYang) [inline mesosturbo/main.go:32]: all the vmt meta fields are set from flag? Then the command to start mesosturbo will be very very long.

**Review by DongyiYang — CHANGES_REQUESTED** [review]

File: communicator/probe/node_probe.go (hunk 1 above)
Comment (pamelasanchezvi) [reply communicator/probe/node_probe.go:177]: we decided we don't support those constraints yet.

File: communicator/metadata/vmt_meta.go (hunk 2 above)
Comment (pamelasanchezvi) [reply communicator/metadata/vmt_meta.go]: right now it's for any connection , i'll change the name

File: communicator/util/types.go (hunk 3 above)
Comment (pamelasanchezvi) [reply communicator/util/types.go:16]: it's for utilization

File: communicator/vmturbocommunicator/vmt_communicator.go (hunk 7 above)
Comment (pamelasanchezvi) [reply communicator/vmturbocommunicator/vmt_communicator.go:169]: it is possible to create container without disk or disk=0

**Review by DongyiYang — APPROVED** [review]

**Pull request merged**